controllers/podchaos/containerkill/containerkill_suite_test.go

```diff
@@ -0,0 +1,96 @@
+package containerkill_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	"github.com/pingcap/chaos-mesh/api/v1alpha1"
+	controllerkill "github.com/pingcap/chaos-mesh/controllers/podchaos/containerkill"
+	. "github.com/pingcap/chaos-mesh/controllers/test"
+	"github.com/pingcap/chaos-mesh/pkg/mock"
+)
+
+func TestContainerKill(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	RunSpecsWithDefaultAndCustomReporters(t,
+		"ContainerKill Suite",
+		[]Reporter{envtest.NewlineReporter{}})
+}
+
+var _ = BeforeSuite(func(done Done) {
+	logf.SetLogger(zap.LoggerTo(GinkgoWriter, true))
+	close(done)
+}, 60)
+
+var _ = AfterSuite(func() {
+})
+
+var _ = Describe("PodChaos", func() {
+	Context("ContainerKill", func() {
+		mock.With("MockChaosDaemonClient", &MockChaosDaemonClient{})
```

Comment:
It's better to call the finalizer at the end of the test function. Otherwise, the mocked value will leak to other tests unexpectedly.
For example:
```
// Assume we have a function named `f` to be tested:
func f() error {
    if err := mock.On("MockF"); err != nil {
        return err.(error)
    }
    return nil
}

Context("f_should_error", func() {
    mock.With("MockF", errors.New("some error"))
    err := f()
    Expect(err).To(HaveOccurred())
})

Context("f_should_not_error", func() {
    err := f()
    Expect(err).ToNot(HaveOccurred())
})
```
Because we didn't call finiazer at the end if the first test  function(f_should_error),  the f will still be mocked at the second test function (f_should_not_error).  So the f will still return error in f_should_not_error, what is unexpected.

Use `defer mock.With("MockChaosDaemonClient", &MockChaosDaemonClient{})()` to call finalizer at the end of the test function automaticly.

Reply:
I see, thanks!